x-pack/platform/plugins/shared/task_manager/server/task_scheduling.ts

```diff
@@ -207,32 +210,40 @@ export class TaskScheduling {
    * `schedule` and `runAt` will be recalculated after task run finishes
    *
    * @param {string[]} taskIds  - list of task ids
-   * @param {IntervalSchedule} schedule  - new schedule
+   * @param {IntervalSchedule | RruleSchedule} schedule  - new schedule
    * @returns {Promise<BulkUpdateTaskResult>}
    */
   public async bulkUpdateSchedules(
     taskIds: string[],
-    schedule: IntervalSchedule
+    schedule: IntervalSchedule | RruleSchedule
   ): Promise<BulkUpdateTaskResult> {
     return retryableBulkUpdate({
       taskIds,
       store: this.store,
       getTasks: async (ids) => await this.bulkGetTasksHelper(ids),
-      filter: (task) =>
-        task.status === TaskStatus.Idle && task.schedule?.interval !== schedule.interval,
+      filter: (task) => task.status === TaskStatus.Idle && !_.isEqual(task.schedule, schedule),
       map: (task) => {
-        const oldIntervalInMs = parseIntervalAsMillisecond(task.schedule?.interval ?? '0s');
+        const { rrule, interval } = schedule || {};
+        let nextRunAt: number | undefined;
 
-        // computing new runAt using formula:
-        // newRunAt = oldRunAt - oldInterval + newInterval
-        const newRunAtInMs = Math.max(
-          Date.now(),
-          task.runAt.getTime() - oldIntervalInMs + parseIntervalAsMillisecond(schedule.interval)
-        );
+        if (interval) {
+          nextRunAt = task.scheduledAt.getTime() + parseIntervalAsMillisecond(interval);
+        } else if (rrule) {
+          const _rrule = new RRule({
+            ...rrule,
+            dtstart: task.scheduledAt,
+          });
+
+          // adding 1ms for the slim chance that scheduled.At === now
+          nextRunAt = _rrule.after(new Date(Date.now() + 1))?.getTime();
+        }
+
+        const newRunAtInMs = Math.max(Date.now(), nextRunAt!);
 
         return { ...task, schedule, runAt: new Date(newRunAtInMs) };
       },
       validate: false,
+      mergeAttributes: false,
```

Comment:
This defaults to `true` everywhere.

I added this parameter for a scenario where we were trying to update from one schedule type to another. The functional tests were showing that both values were being saved in ES.

To avoid that, I introduced this parameter and only set it here where the `schedule` and `runAt` values are updated.

My only concern is with `task`. It should be the same. Could anyone confirm?